botocore/credentials.py

```diff
@@ -67,7 +67,7 @@ def create_credential_resolver(session, cache=None):
     env_provider = EnvProvider()
     container_provider = ContainerProvider()
     instance_metadata_provider = InstanceMetadataProvider(
-        iam_role_fetcher=InstanceMetadataFetcher(
+        iam_role_fetcher=InstanceMetadataCredentialFetcher(
```

Comment:
I'm a little concerned about this impacting people.  I'd prefer we keep the name the same, and just create something new like `IMDSClient` for making arbitrary requests to IMDS.  We can revisit the names in a major version bump.

Reply:
Ah good point, I can change the names around.